django/thunderstore/social/api/experimental/views/complete_login.py

```diff
@@ -48,10 +56,87 @@ def post(self, request: Request, provider: str) -> HttpResponse:
         if not helper_class:
             return Response("Unsupported OAuth provider", HTTP_400_BAD_REQUEST)
 
-        # TODO: get or create a user account based on user_info, create
-        # a new session for the user and return session id.
+        # TODO: create a new session for the user and return session id.
         helper = helper_class(code, redirect_uri)
         helper.complete_login()
         user_info = helper.get_user_info()
+        user = get_or_create_auth_user(provider, user_info)
 
         return Response({"session_id": "TODO"})
+
+
+@transaction.atomic
+def get_or_create_auth_user(provider: str, user_info: UserInfoSchema) -> DjangoUser:
+    """
+    Return local user object based on OAuth user data.
+
+    The "local" user object is the one defined in Django config. We keep
+    additional OAuth related data in a separate table, which is a relic
+    from using Social Auth library to handle the authentication.
+    """
+
+    with connection.cursor() as cursor:
+        cursor.execute(
```

Comment:
Any reason why we're doing raw SQL instead of relying on [the model class](https://github.com/python-social-auth/social-app-django/blob/master/social_django/models.py#L65)?

Reply:
I assumed we'll drop Social Auth from dependencies once we don't actively use it anymore, so the new implementation doesn't depend on it.

Reply:
Yeah but the table is managed by social auth so it'd make sense to use the model class for the time being. Once we drop social auth from dependencies I'd see us migrating to a new table entirely which is managed by some of our own models.